### PR TITLE
ID-267 Rip out opendj

### PR DESCRIPTION
**Pull request by Ghost-in-a-Jar**
Ticket: <Link to Jira ticket>
<Put notes here to help reviewer understand this PR>

---

**PR checklist**

- [ ] I've followed [the instructions](https://github.com/broadinstitute/sam/blob/develop/CONTRIBUTING.md#api-changes) if I've made any changes to the API, _especially_ if they're breaking changes
- [ ] I've filled out the [Security Risk Assessment](https://broadinstitute.github.io/dsp-appsec-security-risk-assessment/) and attached the result to the JIRA ticket

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -63,17 +58,9 @@ jobs:
           ./minnie-kenny.sh --force
           git secrets --scan-history
 
-      - name: Run JndiSchemaInit tests
-        id: tests
-        run: sbt clean "testOnly -- -n org.broadinstitute.tags.SchemaInit"
-
       - name: Run unit tests and generate coverage report
         id: coverageReport
-        run: sbt coverage "testOnly -- -l org.broadinstitute.tags.SchemaInit" coverageReport
```
Comment (Ghost-in-a-Jar) [inline .github/workflows/build.yml:73]: flag

File: docker/build.sh [hunk 2]
```diff
@@ -93,22 +91,17 @@ function make_jar()
...
-    echo $OPENDJ
 
     # Get the last commit hash of the model directory and set it as an environment variable
     GIT_MODEL_HASH=$(git log -n 1 --pretty=format:%h)
 
     # make jar.  cache sbt dependencies.
-    set +e # Turn off error detection so that opendj has a chance to get stopped before exiting
-    docker run --rm --link postgres:postgres --link $OPENDJ:opendj -e DIRECTORY_URL=$DIRECTORY_URL -e GIT_MODEL_HASH=$GIT_MODEL_HASH -e DIRECTORY_PASSWORD=$DIRECTORY_PASSWORD -v $PWD:/working -v jar-cache:/root/.ivy -v jar-cache:/root/.ivy2 hseeberger/scala-sbt:eclipse-temurin-17.0.2_1.6.2_2.13.8 /working/docker/init_schema.sh /working
-    docker restart $OPENDJ
+    docker run --rm --link postgres:postgres -e DIRECTORY_URL=$DIRECTORY_URL -e GIT_MODEL_HASH=$GIT_MODEL_HASH -e DIRECTORY_PASSWORD=$DIRECTORY_PASSWORD -v $PWD:/working -v jar-cache:/root/.ivy -v jar-cache:/root/.ivy2 hseeberger/scala-sbt:eclipse-temurin-17.0.2_1.6.2_2.13.8 /working/docker/init_schema.sh /working
```
Comment (Ghost-in-a-Jar) [inline docker/build.sh]: remove DIRECTORY_*

File: src/main/scala/org/broadinstitute/dsde/workbench/sam/Boot.scala [hunk 3]
```diff
@@ -179,41 +137,28 @@ object Boot extends IOApp with LazyLogging {
             appConfig.adminConfig
           )
           val googleGroupSynchronizer =
-            new GoogleGroupSynchronizer(backgroundDirectoryDAO, backgroundAccessPolicyDAO, cloudExtension.googleDirectoryDAO, cloudExtension, resourceTypeMap)(
-              backgroundLdapExecutionContext)
+            new GoogleGroupSynchronizer(backgroundDirectoryDAO, backgroundAccessPolicyDAO, cloudExtension.googleDirectoryDAO, cloudExtension, resourceTypeMap)
```
Comment (gpolumbo-broad) [inline src/main/scala/org/broadinstitute/dsde/workbench/sam/Boot.scala]: TODO:  Do we need a separate execution context here?

File: src/main/scala/org/broadinstitute/dsde/workbench/sam/service/StatusService.scala [hunk 4]
```diff
@@ -6,59 +6,38 @@ import akka.util.Timeout
 import cats.effect.IO
 import cats.effect.unsafe.implicits.global
 import com.typesafe.scalalogging.LazyLogging
-import org.broadinstitute.dsde.workbench.sam.dataAccess.{ConnectionType, DirectoryDAO, RegistrationDAO}
+import org.broadinstitute.dsde.workbench.sam.dataAccess.{DirectoryDAO}
 import org.broadinstitute.dsde.workbench.sam.db.DbReference
 import org.broadinstitute.dsde.workbench.sam.util.SamRequestContext
 import org.broadinstitute.dsde.workbench.util.health.HealthMonitor.GetCurrentStatus
-import org.broadinstitute.dsde.workbench.util.health.Subsystems.{Database, OpenDJ, Subsystem}
```
Comment (andy7i) [inline src/main/scala/org/broadinstitute/dsde/workbench/sam/service/StatusService.scala:13]: TODO: create ticket to remove this from workbench-libs subsystems

File: src/test/scala/org/broadinstitute/dsde/workbench/sam/api/StatusRouteSpec.scala [hunk 5]
```diff
@@ -38,29 +35,24 @@ class StatusRouteSpec extends AnyFlatSpec with Matchers with ScalatestRouteTest
     implicit val patienceConfig = PatienceConfig(timeout = 1 second)
     eventually {
       Get("/status") ~> samRoutes.route ~> check {
-        responseAs[StatusCheckResponse] shouldEqual StatusCheckResponse(true, Map(OpenDJ -> HealthMonitor.OkStatus, Database -> HealthMonitor.OkStatus))
```
Comment (andy7i) [inline src/test/scala/org/broadinstitute/dsde/workbench/sam/api/StatusRouteSpec.scala:41]: todo: confirm that it's ok to remove opendj from the status endpoint

File: src/test/scala/org/broadinstitute/dsde/workbench/sam/dataAccess/MockDirectoryDAO.scala [hunk 6]
```diff
@@ -292,15 +288,13 @@ class MockDirectoryDAO(val groups: mutable.Map[WorkbenchGroupIdentity, Workbench
...
-      user <- users.get(userId)
-    } yield {
-      users += user.id -> user.copy(azureB2CId = Option(b2CId))
-    }
-  }
+  override def setUserAzureB2CId(userId: WorkbenchUserId, azureB2CId: AzureB2CId, samRequestContext: SamRequestContext)
+  : IO[Unit] = users.get(userId) match {
+        case Some(user) => IO.pure(users += userId -> user.copy(azureB2CId = Option(azureB2CId)))
+        case None => IO.unit
+      }
```
Comment (andy7i) [inline src/test/scala/org/broadinstitute/dsde/workbench/sam/dataAccess/MockDirectoryDAO.scala:298]: todo: find usages of this method and run the tests comparing the old and new implementation

**Review by andy7i — COMMENTED** [review]
WIP: mob review

File: src/test/scala/org/broadinstitute/dsde/workbench/sam/service/ManagedGroupServiceSpec.scala [hunk 7]
```diff
@@ -55,7 +47,6 @@ class ManagedGroupServiceSpec extends AnyFlatSpec with Matchers with TestSupport
 
   override protected def beforeAll(): Unit = {
     super.beforeAll()
-    runAndWait(schemaDao.init())
   }
```
Comment (Ghost-in-a-Jar) [inline src/test/scala/org/broadinstitute/dsde/workbench/sam/service/ManagedGroupServiceSpec.scala:51]:
```suggestion
```

Comment (Ghost-in-a-Jar) [inline src/test/scala/org/broadinstitute/dsde/workbench/sam/service/ManagedGroupServiceSpec.scala:51]:
```suggestion

```

File: src/test/scala/org/broadinstitute/dsde/workbench/sam/service/ManagedGroupServiceSpec.scala [hunk 8]
```diff
@@ -55,7 +47,6 @@ class ManagedGroupServiceSpec extends AnyFlatSpec with Matchers with TestSupport
...
```
Comment (Ghost-in-a-Jar) [inline src/test/scala/org/broadinstitute/dsde/workbench/sam/service/ManagedGroupServiceSpec.scala]: TODO: delete this beforeAll block

File: src/test/scala/org/broadinstitute/dsde/workbench/sam/service/ResourceServiceSpec.scala [hunk 9]
```diff
@@ -106,7 +98,6 @@ class ResourceServiceSpec extends AnyFlatSpec with Matchers with ScalaFutures wi
 
   override protected def beforeAll(): Unit = {
     super.beforeAll()
-    runAndWait(schemaDao.init())
   }
```
Comment (Ghost-in-a-Jar) [inline src/test/scala/org/broadinstitute/dsde/workbench/sam/service/ResourceServiceSpec.scala]: TODO: remove beforeall block

File: src/test/scala/org/broadinstitute/dsde/workbench/sam/service/StatusServiceSpec.scala [hunk 10]
```diff
@@ -73,16 +56,15 @@ class StatusServiceSpec extends AnyFreeSpec with Matchers with BeforeAndAfterAll
...
     // background database configured to connect to non existent database
     val dbReferenceOverride = DbReference(DatabaseNames.Background, TestSupport.blockingEc)
-    val service = new StatusService(directoryDAOWithAllUsersGroup(false), createLdapDaoWithConnectionPool(), NoExtensions, dbReferenceOverride, pollInterval = 10 milliseconds)
+    val service = new StatusService(directoryDAOWithAllUsersGroup(false), NoExtensions, dbReferenceOverride, pollInterval = 10 milliseconds)
     service
   }
 
   val cases = {
     Seq(
-      ("ok", ok, StatusCheckResponse(true, Map(OpenDJ -> SubsystemStatus(true, None), GoogleGroups -> SubsystemStatus(true, None), Database -> SubsystemStatus(true, None)))),
```
Comment (Ghost-in-a-Jar) [inline src/test/scala/org/broadinstitute/dsde/workbench/sam/service/StatusServiceSpec.scala:82]: TODO: might need to revert this if devops wants us to maintain opendj status endpoint api

File: .github/workflows/build.yml (hunk 1 above)
Comment (Shakespeared) [reply .github/workflows/build.yml:73]: resolved this by removing the tag

File: src/main/scala/org/broadinstitute/dsde/workbench/sam/service/StatusService.scala (hunk 4 above)
Comment (gpolumbo-broad) [reply src/main/scala/org/broadinstitute/dsde/workbench/sam/service/StatusService.scala:13]: https://broadworkbench.atlassian.net/browse/ID-268

File: src/test/scala/org/broadinstitute/dsde/workbench/sam/api/StatusRouteSpec.scala (hunk 5 above)
Comment (gpolumbo-broad) [reply src/test/scala/org/broadinstitute/dsde/workbench/sam/api/StatusRouteSpec.scala:41]: https://broadinstitute.slack.com/archives/CADM7MZ35/p1663357621764979

File: src/test/scala/org/broadinstitute/dsde/workbench/sam/dataAccess/MockDirectoryDAO.scala (hunk 6 above)
Comment (gpolumbo-broad) [reply src/test/scala/org/broadinstitute/dsde/workbench/sam/dataAccess/MockDirectoryDAO.scala:298]: Tested an reverted

**Comment by andy7i** [comment]
Mob review of this PR happened on 9/16/2022 and PR is done and thumbed. We need to update firecloud-develop to remove opendj before we can merge: https://broadworkbench.atlassian.net/browse/ID-269

**Review by Shakespeared — APPROVED** [review]
👍

**Review by andy7i — APPROVED** [review]
Note: We are required to add a ticket # to the title (for compliance reasons)!

this was mob-reviewed 👍

**Comment by rtitle** [comment]
I didn't review closely, but looks great overall! Let's make sure to double check OpenDJ audit logs in dev/prod to make sure traffic has stopped, before deploying this.

**Review by gpolumbo-broad — APPROVED** [review]

**Review by tlangs — APPROVED** [review]

**Pull request merged**